src/components/markdown-renderer/replace-components/markmap/markmap-frame.tsx

```diff
@@ -0,0 +1,27 @@
+import { transform } from 'markmap-lib/dist/transform'
+import { Markmap } from 'markmap-lib/dist/view'
+import React, { Fragment, useEffect, useRef } from 'react'
+
+export interface MarkmapFrameProps {
+  code: string
+}
+
+export const MarkmapFrame: React.FC<MarkmapFrameProps> = ({ code }) => {
+  const diagramContainer = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    if (!diagramContainer.current) {
+      return
+    }
+    const svg: SVGSVGElement = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
```

Comment:
How about letting react create the svg and put it into the ref?

Reply:
Unfortunately that can't be done, because the lib wants a `SVGElement`, but the ref for a svg in React needs to be a `SVGSVGElement` and those do not mix. I even tried using `as` as seen below:
![grafik](https://user-images.githubusercontent.com/3978662/93681407-74db3d00-faae-11ea-8b3f-f7bc0d43d3a1.png)

Reply:
did you try svgRef.current ?

Reply:
oh yeah that worked at least on the code side.

Reply:
now every time the code changed i get a new svg added to the dom and the old one persists

Reply:
![grafik](https://user-images.githubusercontent.com/3978662/93684518-a6083d00-faaf-11ea-9909-59a62681c894.png)

Reply:
Okay. Forget my idea. It would have been nice if it worked, but hell no.